.circleci/config.yml

```diff
@@ -1,21 +1,73 @@
-version: 2
+version: 2.1
+
+orbs:
+  node: circleci/node@4.4.0
+
+workflows:
+  version: 2
+  workflow:
+    jobs:
+      - build:
+          matrix:
+            parameters:
+              # We should test against "latest Node 16", but there's an issue with 16.2
+              # This will be updated soon, along with a bug report to Node.js
+              # node-version: ["lts/erbium", "lts/fermium", "16"]
+              node-version: ["lts/erbium", "lts/fermium", "16.1"]
+      - release:
+          requires:
+            - build
+          filters:
+            branches:
+              only:
+                - main
+
 jobs:
   build:
-    docker:
-      - image: circleci/node:14.8.0
+    parameters:
+      node-version:
+        description: Node version to run tests under (set by matrix build config)
+        type: string
+    machine:
+      image: ubuntu-2004:202104-01
```

Comment:
Instead of doing the docker-in-docker trick, we're now using a "machine" image (VM) so we don't have to build a tests docker container.